src/rules/selector-max-specificity/index.js

```diff
@@ -35,15 +35,21 @@ export default function (max) {
         resolvedNestedSelector(selector, rule).forEach(resolvedSelector => {
           // Return early if selector contains a not pseudo-class
           if (selector.indexOf(":not(") !== -1) { return }
+          // Return early if selector contains a matches
+          if (selector.indexOf(":matches(") !== -1) { return }
```

Comment:
> because they are not going to know if the string occurs within a comment or string.

Isn't `rule.selectors` [stripped of comments](https://astexplorer.net/#/LSJWUEhoKv)?

> My suggestion is that we drop them and rely on the try-catch.

If rely on the try-catch, won't a PostCSS warning be produced for each `:not` and `:matches` pseudo-classes in a source when this rule is turned off? Explicitly checking for and ignoring these *known* pseudo-classes seems less abrasive. The try-catch is then only fallen back on when something *unknown* causes the parser in the `specificity` module to break.  Or have I misunderstood the output of the try-catch?